src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,38 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-/**
- * Parameterize this service and add its implementation in a separate class.
- */
-public interface MachineService {
+public interface MachineService<T extends Machine> {
+    List<Machine> getAll(Class<? extends Machine> type);
```

Comment:
We need to use introduced parameter in our methods, otherwise it makes no sense to make it
```suggestion
    List<Machine> getAll(Class<? extends T> type);
```